docs/rackhd/tftp_dhcp_server.rst

```diff
@@ -0,0 +1,313 @@
+By default, RackHD use `on-tftp`_ for TFTP service, `ISC DHCP Server`_ and DHCP proxy `on-dhcp-proxy`_ for DHCP service,
+and they are deployed in RackHD server along with other RackHD service `on-http`, `on-taskgraph`, `on-syslog`.
+They could be replaced with other TFTP and DHCP services, and also could be deployed to another separated server.
+
++---------------------------+---------------------------------------------+-------------------------------------------------+
+| Cases                     |            Supported TFTP Service           |             Supported DHCP Service              |
++===========================+=============================================+=================================================+
+| TFTP and DHCP service are | 1. on-tftp(default)                         | 1. ISC DHCP + on-dhcp-proxy(default)            |
+| in RackHD server          | 2. Third-party TFTP service such as         | 2. ISC DHCP only                                |
+|                           |    in.tftpd(tftp-hpa) in Ubuntu OS          | 3. Third-party DHCP Service + DHCP proxy        |
+|                           |                                             | 4. Third-party DHCP Service only                |
++---------------------------+---------------------------------------------+-------------------------------------------------+
+| TFTP and DHCP service are | 1. on-tftp                                  | 1. ISC DHCP + on-dhcp-proxy                     |
+| in separate server        | 2. Third-party TFTP service such as         | 2. ISC DHCP only                                |
+|                           |    in.tftpd(tftp-hpa) in Ubuntu OS          | 3. Third-party DHCP Service + DHCP proxy        |
+|                           |                                             | 4. Third-party DHCP Service only                |
++---------------------------+---------------------------------------------+-------------------------------------------------+
+
+**NOTE:** "Third-party" service means it's not RackHD default service.
+
+TFTP and DHCP are in RackHD Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+TFTP Service Configuration in RackHD Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Default on-tftp Configuration
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+RackHD default TFTP service is `on-tftp`_, it could be configured by fields `tftpBindAddress`, `tftpBindPort`, `tftpRoot`
+in `config.json`_, and `RackHD iPXE`_ files are put to `tftpRoot` directory.
+
+.. code-block:: javascript
+
+    ...
+    "tftpBindAddress": "172.31.128.1",
+    "tftpBindPort": 69,
+    "tftpRoot": "./static/tftp",
+    ...
+
+Third-Party TFTP Service Configuration
+""""""""""""""""""""""""""""""""""""""
+
+.. _RackHD TFTP Templates: https://github.com/RackHD/on-tftp/tree/master/data/templates
+
+Any other TFTP service could be used by RackHD, just put `RackHD iPXE`_ files to TFTP root directory.
```

Comment:
Perhaps better wording, instead of "Any other TFTP service could be used by RackHD":

"In many cases, another TFTP service can used with RackHD. RackHD simply needs the files that on-tftp would serve to be provided by another instance of TFTP. You can frequently do this by simply placing the `RackHD iPXE`_ files into the TFTP service root directory."